include/aws_iot_mqtt_client_interface.h

```diff
@@ -58,154 +58,242 @@ extern "C" {
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
  */
+/* @[declare_mqtt_publish] */
 IoT_Error_t aws_iot_mqtt_publish(AWS_IoT_Client *pClient, const char *pTopicName, uint16_t topicNameLen,
 								 IoT_Publish_Message_Params *pParams);
+/* @[declare_mqtt_publish] */
 
 /**
  * @brief Subscribe to an MQTT topic.
  *
- * Called to send a subscribe message to the broker requesting a subscription
- * to an MQTT topic.
- * @note Call is blocking.  The call returns after the receipt of the SUBACK control packet.
- * @warning pTopicName and pApplicationHandlerData need to be static in memory.
+ * This function sends an MQTT subscribe packet to the server. It registers
+ * a subscription that will cause the provided callback function to be invoked
+ * when the server sends a matching message to the client.
  *
- * @param pClient Reference to the IoT Client
- * @param pTopicName Topic Name to publish to. pTopicName needs to be static in memory since
- * no malloc is performed by the SDK
- * @param topicNameLen Length of the topic name
- * @param qos Quality of service for subscription
- * @param pApplicationHandler Reference to the handler function for this subscription
- * @param pApplicationHandlerData Point to data passed to the callback. pApplicationHandlerData
- * also needs to be static in memory since no malloc is performed by the SDK
+ * @note Incoming messages are handled by @ref mqtt_function_yield. Therefore,
+ * @ref mqtt_function_yield must always be called regularly if any subscriptions
+ * are active.
  *
- * @return An IoT Error Type defining successful/failed subscription
+ * @param[in] pClient MQTT client context
+ * @param[in] pTopicName Topic for subscription
+ * @param[in] topicNameLen Length of topic
+ * @param[in] qos Quality of service for subscription
+ * @param[in] pApplicationHandler Callback function for incoming messages that arrive
+ * on this subscription.
+ * @param[in] pApplicationHandlerData Data passed to the callback.
+ *
+ * @return `IoT_Error_t`: See `aws_iot_error.h`
+ *
+ * @attention The `pTopicName` parameter is not copied. It must remain valid for the duration
+ * of the subscription (until @ref mqtt_function_unsubscribe) is called.
  */
+/* @[declare_mqtt_subscribe] */
 IoT_Error_t aws_iot_mqtt_subscribe(AWS_IoT_Client *pClient, const char *pTopicName, uint16_t topicNameLen,
 								   QoS qos, pApplicationHandler_t pApplicationHandler, void *pApplicationHandlerData);
+/* @[declare_mqtt_subscribe] */
 
 /**
- * @brief Subscribe to an MQTT topic.
+ * @brief Resubscribe to topic filter subscriptions in a previous MQTT session.
  *
- * Called to resubscribe to the topics that the client has active subscriptions on.
- * Internally called when autoreconnect is enabled
+ * This function restores subscriptions that were previously present in an
+ * MQTT session. Its primary use is to restore subscriptions after a session
+ * is manually disconnected and reopened.
  *
```

Comment:
The previous documentation mentioned "_Internally called when autoreconnect is enabled_". Why was this line removed?

Reply:
Because I don't think the internal details of the library should be mentioned in API-level documentation.

Reply:
Okay, valid point.